File: lib/models/exchangerate.model.dart
class ExchangeRateModel {
  int? id;
  String serverID;
  String bitcoinUnit;
  String fiatCurrency;
  String sign;
  String exchangeRateTime;
  int exchangeRate;
  int cents;

  ExchangeRateModel({
    required this.id,
    required this.serverID,
    required this.bitcoinUnit,
    required this.fiatCurrency,
    required this.sign,
    required this.exchangeRateTime,
    required this.exchangeRate,
    required this.cents,
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'serverID': serverID,
      'bitcoinUnit': bitcoinUnit,
      'fiatCurrency': fiatCurrency,
      'sign': sign,
      'exchangeRateTime': exchangeRateTime,
      'exchangeRate': exchangeRate,
      'cents': cents,
    };
  }

  factory ExchangeRateModel.fromMap(Map<String, dynamic> map) {
    return ExchangeRateModel(
      id: map['id'],
      serverID: map['serverID'],
      bitcoinUnit: map['bitcoinUnit'],
      fiatCurrency: map['fiatCurrency'],
      sign: map['sign'],
      exchangeRateTime: map['exchangeRateTime'],
      exchangeRate: map['exchangeRate'],
      cents: map['cents'],
    );
  }
}

File: rust/src/proton_wallet/db/model/exchange_rate_model.rs
use rusqlite::{Result, Row};
use serde::{Deserialize, Serialize};
use serde_rusqlite::from_row;

use super::model::ModelBase;

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct ExchangeRateModel {
    pub id: Option<u32>,
    pub server_id: String,
    pub bitcoin_unit: String,
    pub fiat_currency: String,
    pub sign: String,
    pub exchange_rate_time: String,
    pub exchange_rate: u32,
    pub cents: u32,
}

impl ModelBase for ExchangeRateModel {
    fn from_row(row: &Row) -> Result<Self> {
        Ok(from_row::<ExchangeRateModel>(row).unwrap())
    }
}
